### PR TITLE
AIP-6005 Use PVC as tmp path in S3()

### PR DESCRIPTION
**Pull request by talebzeghmi**
* depends on KF Profiles addition of `METAFLOW_ARTIFACT_LOCALROOT` env variable in https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/aip-infrastructure/kubeflow-k8s-profiles/-/merge_requests/530

### CODE REVIEW

File: metaflow/datatools/s3.py [hunk 1]
```diff
@@ -309,6 +309,10 @@ def __init__(
 
         if not boto_found:
             raise MetaflowException("You need to install 'boto3' in order to use S3.")
+        
+        if tmproot is None:
+            artifact_localroot = from_conf("METAFLOW_ARTIFACT_LOCALROOT")
+            tmproot = artifact_localroot if artifact_localroot else "."
```
Comment (cloudw) [inline metaflow/datatools/s3.py:315]: If no PVC is mounted would this path still be valid as an ephemeral storage path?

**Review by cloudw — COMMENTED** [review]
Is the file `src/kfp` expected in this PR?

File: metaflow/datatools/s3.py (hunk 1 above)
Comment (talebzeghmi) [reply metaflow/datatools/s3.py:315]: yes that is where the logs are put anyhow:

https://github.com/zillow/metaflow/blob/376ef787388fd3942986b2db25f819a3922b1ef9/metaflow/plugins/kfp/kfp_constants.py#L24

**Comment by cloudw** [comment]
Is the file `src/kfp` expected in this PR?

**Comment by talebzeghmi** [comment]
> Is the file src/kfp expected in this PR?
not at all! nice catch! This must have been added by VSCode PR creation.

**Review by cloudw — APPROVED** [review]

**Pull request merged**